pkg/kapp/resources/resource_filter.go

```diff
@@ -92,6 +94,26 @@ func (f ResourceFilter) Matches(resource Resource) bool {
 		}
 	}
 
+	if len(f.Labels) > 0 {
+		var matched bool
+		for _, label := range f.Labels {
+			labelSelector, err := labels.Parse(label)
+			expectedLabelKey, expectedLabelVal, err := NewSimpleLabel(labelSelector).KV()
+			if err != nil {
+				return false
+			}
+			if val, found := resource.Labels()[expectedLabelKey]; found {
+				if val == expectedLabelVal {
+					matched = true
+					break
+				}
+			}
```

Comment:
instead of doing manual matching, lets use https://github.com/vmware-tanzu/carvel-kapp/blob/e0bf1c9d485c5c59a23b4cfe75915782e7fb4b42/vendor/k8s.io/apimachinery/pkg/labels/selector.go#L38. this gives us ability to work with label selectors like: `!key`, `key=foo`, `key!=foo` etc.

Reply:
done. Please check